lib/pbench/server/auth/__init__.py

```diff
@@ -74,22 +83,21 @@ def add_header_param(self, key: str, value: str):
         self.headers[key] = value
 
     def del_param_headers(self, key: str):
-        """Remove a specific header parameter.
+        """
+        Remove a specific header parameter.
         Args:
             Key to delete from the headers.
         """
         del self.headers[key]
 
     def set_well_known_endpoints(self):
-        """Sets the well-known configuration endpoints for the current oidc client.
-        This includes common useful endpoints for decoding tokens, getting user
-        information etc.
+        """
+        Sets the well-known configuration endpoints for the current OIDC
+        client. This includes common useful endpoints for decoding tokens,
+        getting user information etc.
         """
         well_known_endpoint = "/.well-known/openid-configuration"
-        if self.realm_name:
-            well_known_uri = f"{self.server_url}{self.realm_name}{well_known_endpoint}"
-        else:
-            well_known_uri = f"{self.server_url}{well_known_endpoint}"
+        well_known_uri = f"{self.server_url}{self.realm_name}{well_known_endpoint}"
```

Comment:
I assume this is why you changed the default realm name, to avoid the conditional? But do we really need/want to support an empty realm name? Is this a realistic mode of operation? If not, maybe we should just be failing if it's `None` instead of providing an empty string that (probably) won't do what anyone really wants?

Reply:
As far as I know, the Google OIDC client provides an auth server URL that can be used to fetch the well-known endpoints. Their [documentation](https://developers.google.com/identity/protocols/oauth2/openid-connect#python_1) mentions the following
> To use Google's OpenID Connect services, you should hard-code the Discovery-document URI (https://accounts.google.com/.well-known/openid-configuration) into your application

So in this case the server URI would be `https://accounts.google.com/` and realm-name is `""`

Reply:
Well, that's pretty convincing...provided, of course, that we need or want to keep the base-URL and the realm separate from each other.  I don't see them used separately, so perhaps we could glue them together and store them that way, and, in the case of Google, no one would be the wiser.

Reply:
I guess I'll resolve this one in passing, although I kinda like Webb's suggestion...